### PR TITLE
#154 Header h3 support

### PR DESCRIPTION
**Pull request by kalbfled**
These changes add support for including level 3 headers as markdown.  i.e. `###` --> `<h3>...</h3>`

For h3, I used the prescribed font-size of 20.8px.  The other styles are a duplicate of h2.  For non-html messages, there is no differentiation between h2 and h3.  (The ticket doesn't prescribe anything.)

For testing, I updated the "header test" template of the "VA Notify" service in Dev and used the portal UI to "send test email."  It works.

image

Note the font-size:
image

Issue #154

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ The `./scripts/run_tests.sh` script runs all unit tests using [py.test](http://p
 
 With the virtual environment active, run `python setup.py --version` to see the current version **on the current branch**.  Use `git tag` to add release tags to commits, and push the tags.
 
-You can reference these release tags in requirements files in other repositories.  See [notification-api](https://github.com/department-of-veterans-affairs/notification-api/blob/master/requirements-app.txt) for an example.
+You can reference these release tags in requirements files in other repositories.  See [notification-api](https://github.com/department-of-veterans-affairs/notification-api/blob/master/requirements-app.txt) for an example.  This isn't strictly necessary anymore because notification-api uses Poetry to manage dependencies.
```
Comment (cris-oddball) [inline README.md:23]: please remove the first sentence - that linked file no longer exists.
please add 'by running the command `poetry update notification-utils` when the utils PR is merged' to the second sentence.

```suggestion
After merging changes, to update the notification-api to use the updates, run `poetry update notification-utils` in an api branch and then push the PR for approval/merge. The PR should just containt chanegs made to the lock file.
```

**Review by cris-oddball — CHANGES_REQUESTED** [review]

**Review by EvanParish — COMMENTED** [review]
It looks good to me once you make the change Cris requested.

File: README.md (hunk 1 above)
Comment (kalbfled) [reply README.md]: Changed.

Comment (cris-oddball) [reply README.md]: well, fix my typos if you take that suggestion.

File: tests/test_formatters.py [hunk 2]
```diff
@@ -324,6 +324,25 @@ def test_level_2_header(markdown_function, expected):
...
+    [
+        notify_plain_text_email_markdown,
+        (
+            '\n'
+            '\ninset text'
+            '\n-----------------------------------------------------------------'
+        ),
+    ],
+))
+def test_level_3_header(markdown_function, expected):
```
Comment (cris-oddball) [inline tests/test_formatters.py:342]: this needs to follow the test naming convention.

**Review by cris-oddball — CHANGES_REQUESTED** [review]

File: tests/test_formatters.py (hunk 2 above)
Comment (kalbfled) [reply tests/test_formatters.py:342]: Existing unit tests are named `test_level_1_header` and `test_level_2_header`.  Is it better to give a new test an inconsistent name?

Comment (cris-oddball) [reply tests/test_formatters.py:342]: Kyle approved the test as it stands.

**Review by cris-oddball — APPROVED** [review]

File: tests/test_formatters.py (hunk 2 above)
Comment (k-macmillan) [reply tests/test_formatters.py:342]: Logic:
Testing guide is kind of vague [for naming](https://github.com/department-of-veterans-affairs/vanotify-team/blob/master/Process/testing_guide.md#test-naming).
> Naming shall be consistent

Could be interpreted as continuing the existing naming convention. In the case of that PR it'd be weird to name it different when it's a copy/paste of the other two. We didn't want to rename existing tests under most circumstances. I think maintaining the name is fine in this case.

**Review by mchlwellman — APPROVED** [review]

**Review by k-macmillan — APPROVED** [review]

**Pull request merged**